src/app/options/qgsoptions.cpp

```diff
@@ -642,16 +642,33 @@ QgsOptions::QgsOptions( QWidget *parent, Qt::WindowFlags fl, const QList<QgsOpti
 
   // set if base unit of measure tool should be changed
   bool baseUnit = mSettings->value( QStringLiteral( "qgis/measure/keepbaseunit" ), true ).toBool();
-  if ( baseUnit )
-  {
-    mKeepBaseUnitCheckBox->setChecked( true );
-  }
+  mKeepBaseUnitCheckBox->setChecked( baseUnit );
+
+  mPlanimetricMeasurementsComboBox->setChecked( mSettings->value( QStringLiteral( "measure/planimetric" ), false, QgsSettings::Core ).toBool() );
+
+  // set the measure tool copy settings
+  connect( mSeparatorOther, &QRadioButton::toggled, mSeparatorCustom, &QLineEdit::setEnabled );
+  bool includeHeader = mSettings->value( QStringLiteral( "qgis/measure/clipboard_header" ), false ).toBool();
```

Comment:
Could you make sure that all settings you are introducing here rely on the new settings framework (here's an example: https://github.com/qgis/QGIS/blob/master/src/app/qgsidentifyresultsdialog.cpp#L102)

Reply:
Sure.

Reply:
Actually `qgis/measure/xxx` keys are used all over the place, so I think it might be out of scope for this PR. The example you gave is really QWebView-oriented.

IIUC, the new settings framework, introduced by @3nids relies on declaring settings entries as `QgsSettingsEntryBool|String|Whatever` 
https://github.com/qgis/QGIS/blob/47d88e94871bac8deaa2026aa6d0057fd862ff81/src/app/elevation/qgselevationprofilewidget.cpp#L59-L61

Reply:
@YoannQDQ , I'm not suggesting you migrate the pre-existing settings keys in this PR :) however I do think any *newly introduced* settings should use the new framework (e.g. src/app/gps makes use of strings, booleans, etc.).

If we keep adding old style settings in new code, we're just increasing the migration burden down the road :)

Reply:
Fair enough. I'll do it.

Reply:
yes, please.
@YoannQDQ can you create a new node directly in QgsSettingsTree, call it `measure` maybe? (having things in `qgis` is not really meaningful). Also, as written in docs, please use kebab-case. Cheers

Reply:
Done